drivers/SMBSR.py

```diff
@@ -108,27 +108,27 @@ def mount(self, mountpoint=None):
         except util.CommandException, inst:
             raise SMBException("Failed to make directory: code is %d" %
                                 inst.code)
+        return mountpoint
+
+    def mount(self, mountpoint=None):
+
+        mountpoint = self.makeMountPoint(mountpoint)
 
-        self.credentials = os.path.join("/tmp", util.gen_uuid())
+        new_env, domain = cifutils.getCIFCredentials(self.dconf, self.session)
 
-        options = self.getMountOptions()
+        options = self.getMountOptions(domain)
```

Comment:
I wonder if there is a further (future?) refinement possible here to combine getting options and environment into a single method that avoids passing the domain back from the getCredentials and then passing it into the getMountOptions. Probably too much for this change so maybe file an enhancement ticket for the future?

Reply:
I agree.
Ive seen a lot of copy and paste between ISOSR and SMBSR, which can be farmed of into a util file as part of any enhancement.

Reply:
As you found out experimentally, the man page is wrong here.

Looking at the source has the comment
/*
		 * Note that the password will not be retrieved from the
		 * USER env variable (ie user%password form) as there is
		 * already a PASSWD environment varaible
		 */"

I have created the ticket CA-330305 for a general cleanup.

Reply:
> As you found out experimentally, the man page is wrong here.
> 
> Looking at the source has the comment
> /*
> * Note that the password will not be retrieved from the
> * USER env variable (ie user%password form) as there is
> * already a PASSWD environment varaible
> */"
> 
> I have created the ticket CA-330305 for a general cleanup.

Indeed, I think the manpage is completely wrong, it doesn't seem to even work with the domain in the user name on the commandline. it probably did work once but the codes been updated without the associated docs.